### PR TITLE
Normalize fuzzer & coverage library reference inconsistencies.

### LINKED ISSUE #1815: Inconsistent use of sanitize-coverage in fuzzers
### Describe the bug

core_helper_fuzzer doesn't use that option, and verifier_fuzzer only does so for release build

```
C:\Users\dthal\git\ebpf\ebpf-for-windows\tests>findstr /s sanitize *proj
libfuzzer\bpf2c\bpf2c_fuzzer.vcxproj:      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
libfuzzer\bpf2c\bpf2c_fuzzer.vcxproj:      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
libfuzzer\execution_context\execution_context_fuzzer.vcxproj:      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
libfuzzer\execution_context\execution_context_fuzzer.vcxproj:      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
libfuzzer\verifier\verifier_fuzzer.vcxproj:      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
```

### OS information

_No response_

### Steps taken to reproduce bug

See above.

### Expected behavior

Use is consistent across fuzzers.

### Actual outcome

Use is inconsistent across fuzzers.

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by gtrevi**
Closes #1815

## Description

This PR:
- Normalizes reference inconsistencies to fuzzer & coverage libraries.
- Uniforms directory names of fuzzer projects.

## Testing

CI/CD pipeline.

### CODE REVIEW

File: tests/libfuzzer/bpf2c_fuzzer/bpf2c_fuzzer.vcxproj [hunk 1]
```diff
@@ -47,30 +47,27 @@
...
+      <AdditionalDependencies>$(FuzzerLibs);mincore.lib;%(AdditionalDependencies)</AdditionalDependencies>
       <AdditionalLibraryDirectories>$(VCToolsInstallDir)lib\$(Platform)</AdditionalLibraryDirectories>
     </Link>
   </ItemDefinitionGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
     <ClCompile>
       <PreprocessorDefinitions>NDEBUG;_CONSOLE;NO_CATCH;%(PreprocessorDefinitions)</PreprocessorDefinitions>
       <AdditionalIncludeDirectories>$(SolutionDir)tests\libfuzzer\include;$(SolutionDir)include;$(SolutionDir)resource;$(SolutionDir)external\ebpf-verifier\external\ELFIO;$(SolutionDir)external\ubpf\vm;$(OutDir);$(SolutionDir)external\ebpf-verifier\src;$(SolutionDir)tools\bpf2c;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
+      <WholeProgramOptimization>false</WholeProgramOptimization>
```
Comment (dthaler) [inline tests/libfuzzer/bpf2c_fuzzer/bpf2c_fuzzer.vcxproj:65]: PR #1931 removed this line

File: tests/libfuzzer/execution_context/execution_context_fuzzer.vcxproj [hunk 2]
```diff
@@ -55,25 +55,26 @@
...
       <AdditionalIncludeDirectories>$(SolutionDir)tests\libfuzzer\include;$(SolutionDir)libs\api_common;$(SolutionDir)include;$(SolutionDir)libs\api;$(SolutionDir)libs\ebpfnetsh;$(SolutionDir)tests\libs\util;$(SolutionDir)tests\libs\common;$(OutDir);$(SolutionDir)external\ebpf-verifier\src;$(SolutionDir)libs\service;$(SolutionDir)rpc_interface;$(SolutionDir)libs\platform;$(SolutionDir)libs\platform\user;$(SolutionDir)libs\execution_context;$(SolutionDir)tests\end_to_end;$(SolutionDir)tests\sample;$(SolutionDir)tests\sample\ext\inc;$(SolutionDir)\tests\xdp;$(SolutionDir)tools\export_program_info;$(SolutionDir)libs\thunk;$(SolutionDir)libs\thunk\mock;$(SolutionDir)\netebpfext;$(SolutionDir)external\catch2\src;$(SolutionDir)external\catch2\build\generated-includes;$(SolutionDir)external\bpftool;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
+      <WholeProgramOptimization>false</WholeProgramOptimization>
```
Comment (dthaler) [inline tests/libfuzzer/execution_context/execution_context_fuzzer.vcxproj:72]: PR #1931 removed this line

File: tests/libfuzzer/verifier_fuzzer/verifier_fuzzer.vcxproj [hunk 3]
```diff
@@ -56,20 +56,20 @@
...
-      <AdditionalOptions>/fsanitize-coverage=inline-bool-flag /fsanitize-coverage=edge /fsanitize-coverage=trace-cmp /fsanitize-coverage=trace-div %(AdditionalOptions)</AdditionalOptions>
+      <WholeProgramOptimization>false</WholeProgramOptimization>
```
Comment (dthaler) [inline tests/libfuzzer/verifier_fuzzer/verifier_fuzzer.vcxproj:67]: PR #1931 removed this line

File: tools/bpf2c/bpf2c.vcxproj [hunk 4]
```diff
@@ -81,7 +81,8 @@
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
       <OptimizeReferences>true</OptimizeReferences>
       <GenerateDebugInformation>true</GenerateDebugInformation>
-      <AdditionalDependencies>$(SanitizerLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>$(FuzzerLibs);%(AdditionalDependencies)</AdditionalDependencies>
+      <LinkTimeCodeGeneration />
```
Comment (dthaler) [inline tools/bpf2c/bpf2c.vcxproj:85]: PR #1931 removed LinkTimeCodeGeneration

**Review by dthaler — APPROVED** [review]

**Review by Alan-Jowett — APPROVED** [review]

**Pull request merged**